google-cloud-graalvm-samples/google-cloud-graalvm-pubsub-sample/src/main/java/com/example/PubSubSampleApplication.java

```diff
@@ -53,13 +53,16 @@ public static void main(String[] args) throws Exception {
     String topicId = "graal-pubsub-test-" + UUID.randomUUID().toString();
     String subscriptionId = "graal-pubsub-test-sub" + UUID.randomUUID().toString();
 
-    createTopic(topicId);
-    createSubscription(subscriptionId, topicId);
-
-    publishMessage(topicId);
-    subscribeSync(subscriptionId);
-
-    deleteTopic(topicId);
+    try {
+      createTopic(topicId);
+      createSubscription(subscriptionId, topicId);
+
+      publishMessage(topicId);
+      subscribeSync(subscriptionId);
+    } finally {
+      deleteTopic(topicId);
```

Comment:
@anguillanneuf I remember you mentioned that subscriptions should be deleted before their associated topics. Is that still the case?